WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -193,26 +198,19 @@ def register_require_sign_in
   end
 
   def register
-    @competition = competition_from_params
-    @registration = nil
-    @selected_events ||= []
-    if current_user
-      @registration = Registration.find_by(user_id: current_user.id, competition_id: @competition.id) || @competition.registrations.build(user_id: current_user.id)
-      @selected_events = @registration.saved_and_unsaved_events.empty? ? @registration.user.preferred_events : @registration.saved_and_unsaved_events
-    end
+    set_registration_info!
   end
 
   def process_payment
-    competition = competition_from_params
-    if current_user
-      registrations = competition.registrations
-      registration = registrations.find_by_user_id!(current_user.id)
-    end
```

Comment:
I agree, the current code doesn't make any sense. Your refactor also seems like an improvement to me.